astropy/table/table.py

```diff
@@ -3312,6 +3319,17 @@ def insert_row(self, index, vals=None, mask=None):
                             "support masked values".format(col.info.name)
                         )
 
+                if (
```

Comment:
This should not be necessary any more, right?

Reply:
it shouldn't, yes, but apparently I still have broken tests to manage, so I'm keeping it around until I have a working branch and I'm confident I don't need it anymore.